internal/store/leaf/crypto.go

```diff
@@ -56,10 +59,19 @@ func (s *Store) ImportMissingPublicKeys(ctx context.Context, newrs ...string) er
 			debug.Log("Failed to get public key for %s: %s", r, err)
 		}
 
-		if !IsPubkeyUpdate(ctx) && len(kl) > 0 {
```

Comment:
This does change the semantic a bit.
Before we would skip the entire loop if we weren't asked to update keys.

But now only if there are any valid keys for this recipient.
Is that really what we want?

Reply:
the intended change is the following (in pseudocode):
going from:
`if validKeysExist && !wantKeysUpdate {continue}`
to:
`if validKeysExist && (!wantKeysUpdate || bytes.Equals(pubkeyInStore, pubkeyInGPG) ) {continue}`
and the "A or B" part of this condition is moved into the inside of that if block, which only the initial `if validKeysExist` in if statement of that if block.

now that I think about it, the bytewise comparison might not work if there are multiple public keys with that ID. hm…

Reply:
well that last remark was dumb.
I just got caught up on the name `s.crypto.FindRecipients` (which can take more than one key ID) and on `if kl > 0`.
Just to be sure, do you confirm that `kl > 1` shouldn't be possible there?